action.yaml

```diff
@@ -0,0 +1,47 @@
+name: Setup K3d
+description: K3d action with multi-cluster support.
+author: AbsaOSS
+# see: https://haya14busa.github.io/github-action-brandings/
+branding:
+  icon: grid
+  color: red
+inputs:
+  k3d-cluster-name:
+    description: "(Required) Cluster name."
+    required: true
+  k3d-args:
+    description: "(Optional) Additional arguments to k3d cluster. see: https://k3d.io/usage/commands/"
+    required: false
+  k3d-network:
+    description: "(Optional) Cluster network. Default value k3d-action-bridge-network"
+    required: false
+  k3d-subnet:
+    description: "(Optional) Cluster subnet CIDR. Default value 172.16.0.0/24 in case k3d-network is not set."
+    required: false
+  k3d-registry:
+    description: "(Optional) Autogenerate docker registry accessible from the cluster."
+    required: false
+    default: "false"
+  k3d-registry-config-path:
+    description: "(Optional) Path to custom registry configuration file. Variable requires K3D_REGISTRY to be true."
+    required: false
+outputs:
+  k3d-network:
+    description: "Detected k3d version"
+    value: ${{ steps.main.outputs.k3d-network }}
+  k3d-subnet:
+    description: "Detected k3d network subnet"
+    value: ${{ steps.main.outputs.k3d-subnet }}
+runs:
+  using: composite
+  steps:
+    - id: main
+      shell: bash
+      run: ${{ github.action_path }}/run.sh deploy
+      env:
+        K3D_CLUSTER_NAME: ${{ inputs.k3d-cluster-name }}
```

Comment:
well, technically it is k3s cluster but I wouldn't go down that road :D

Reply:
@ytsarev , true. I'll rename

Reply:
@kuritka totally up to you, but yeah it is k3s

Reply:
@somaritane, @ytsarev  What do you think about complete renaming? env vars will refactor as well

k3d-cluster-name -> cluster-name
k3d-args -> args
k3d-network -> network
k3d-subnet -> subnet-CIDR
k3d-registry -> use-default-registry
k3d-registry-config-path -> override-registry-config-path

Reply:
@kuritka looks good to me!